src/Protocol/Relay.php

```diff
@@ -113,13 +113,18 @@ public static function isSolicitedPost(array $tags, string $body, int $authorid,
 		}
 
 		if (!empty($tagList) || !empty($denyTags)) {
+			$content = mb_strtolower(BBCode::toPlaintext($body, false));
+
 			$max_tags = $config->get('system', 'relay_max_tags');
 			if ($max_tags && (count($tags) > $max_tags) && preg_match('/[^@!#]\[url\=.*?\].*?\[\/url\]/ism', $body)) {
-				Logger::info('Possible hashtag spam detected - rejected', ['hashtags' => $tags, 'network' => $network, 'url' => $url, 'causer' => $causer, 'body' => $body]);
-				return false;
-			}
+				$cleaned = preg_replace('/[@!#]\[url\=.*?\].*?\[\/url\]/ism', '', $body);
+				$content_cleaned = mb_strtolower(BBCode::toPlaintext($cleaned, false));
 
-			$content = mb_strtolower(BBCode::toPlaintext($body, false));
+				if (strlen($content_cleaned) < (strlen($content) - strlen($content_cleaned))) {
```

Comment:
You can get away with one fewer `strlen()` call with the following trick:
```suggestion
				if (strlen($content_cleaned) < strlen($content) / 2) {
```

Reply:
No. That's not the same. `a < a -b` is not the same as `a < b / 2`, because `a -b` is not the same as `b / 2`. This is only valid for `a = b`.

Reply:
It works because we know `a < b` since a is `a` part of `b`. So if you want to make sure that `a` is the majority in `b`, you just need to make sure it's bigger than `b / 2`.

Reply:
I still don't really understand the mathematics behind this, but I tested it by brut force:
```
for ($total = 1; $total < 10000; $total++) { 
	for ($cleaned = 0; $cleaned < $total - 1; $cleaned++) { 
		$cond1 = ($cleaned < ($total - $cleaned));
		$cond2 = ($cleaned < ($total / 2));
		if ($cond1 != $cond2) {
			echo 'Total: ' . $total . ' Cleaned: ' . $cleaned . "\n";
			die();
		}
	}
}
```

Reply:
Cool!